subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -577,37 +720,72 @@ int hawkbit_default_config_data_cb(const char *device_id, uint8_t *buffer, const
 				   buffer_size);
 }
 
+#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
+int hawkbit_set_config(struct hawkbit_runtime_config *config)
+{
+	if (k_sem_take(&probe_sem, HAWKBIT_SET_SERVER_TIMEOUT) == 0) {
```

Comment:
Consider K_NO_WAIT instead. Imo it makes sense to call hawkbit_set_config prior to any probe/autoprobe.

Reply:
I have no objection against it. Due to it already being merged, please open a PR to have it changed.